### PR TITLE
EDURED-122: Method for removing all identities

### PR DESCRIPTION
**Pull request by dzolnai**

### CODE REVIEW

File: Sources/TiqrCoreObjC/Classes/IdentityService.m [hunk 1]
```diff
@@ -71,6 +71,28 @@ - (void)deleteIdentityProvider:(IdentityProvider *)identityProvider {
...
 
+- (void)deleteAllIdentitiesAndProviders: (NSError *)error {
+    // Delete the identities first
+    NSFetchRequest *identityRequest = [[NSFetchRequest alloc] initWithEntityName:@"Identity"];
+    NSBatchDeleteRequest *deleteIndentities = [[NSBatchDeleteRequest alloc] initWithFetchRequest:identityRequest];
+
+    [self.persistentStoreCoordinator executeRequest: deleteIndentities
+                                   withContext: self.managedObjectContext
+                                         error: &error];
+    if (error != nil) {
```
Comment (johankool) [inline Sources/TiqrCoreObjC/Classes/IdentityService.m]: executeRequest should return a boolean or a value. Only look at the error if you get back NO or nil. (It's kinda odd, but that's how Obj-C works.)

File: Sources/TiqrCoreObjC/Classes/IdentityService.m [hunk 2]
```diff
@@ -71,6 +71,28 @@ - (void)deleteIdentityProvider:(IdentityProvider *)identityProvider {
     [self deleteObject:identityProvider];
 }
 
+- (void)deleteAllIdentitiesAndProviders: (NSError *)error {
+    // Delete the identities first
+    NSFetchRequest *identityRequest = [[NSFetchRequest alloc] initWithEntityName:@"Identity"];
+    NSBatchDeleteRequest *deleteIndentities = [[NSBatchDeleteRequest alloc] initWithFetchRequest:identityRequest];
```
Comment (johankool) [inline Sources/TiqrCoreObjC/Classes/IdentityService.m]: Typo: indentities -> identities

File: Sources/TiqrCoreObjC/Classes/IdentityService.m [hunk 3]
```diff
@@ -71,6 +71,28 @@ - (void)deleteIdentityProvider:(IdentityProvider *)identityProvider {
...
+
+    [self.persistentStoreCoordinator executeRequest: deleteIndentities
+                                   withContext: self.managedObjectContext
+                                         error: &error];
+    if (error != nil) {
+        return;
+    }
+
```
Comment (johankool) [inline Sources/TiqrCoreObjC/Classes/IdentityService.m]: If the relationship deletion rules of Core Data are setup correctly (big if) you would only need one deletion rule, which would then trigger the deletion of referenced objects.

**Comment by phavekes** [comment]
Why is this change needed? Both Tiqr and eduID do not have a use-case or requirement for deleting all identities?

**Comment by dzolnai** [comment]
> Why is this change needed? Both Tiqr and eduID do not have a use-case or requirement for deleting all identities?

There's only a case for deleting them one-by-one, but not for deleting all of them at once.

File: Sources/TiqrCoreObjC/Classes/IdentityService.m (hunk 3 above)
Comment (dzolnai) [reply Sources/TiqrCoreObjC/Classes/IdentityService.m]: You are right, I tested this and only deleting the identities was enough

**Review by risinggalaxy — APPROVED** [review]

**Review by phavekes — CHANGES_REQUESTED** [review]
If this method is not needed, do not add it

**Comment by phavekes** [comment]
As discussed with @dzolnai : This is not needed

**Pull request closed without merging**